src/main/java/core/basesyntax/FruitTransaction.java

```diff
@@ -0,0 +1,48 @@
+package core.basesyntax;
+
+public class FruitTransaction {
+    private Operation operation;
+    private final String fruit;
+    private final int quantity;
+
+    public FruitTransaction(String operation, String fruit, int quantity) {
+        switch (operation) {
+            case "b" -> this.operation = Operation.BALANCE;
+            case "p" -> this.operation = Operation.PURCHASE;
+            case "r" -> this.operation = Operation.RETURN;
+            case "s" -> this.operation = Operation.SUPPLY;
+            default -> this.operation = null;
+        }
```

Comment:
Make this constructor accept enum itself
for parsing create a dedicated method inside of said enum
don't use switch - use for each + throw exception if not found